core/row/row.selector.js

```diff
@@ -0,0 +1,120 @@
+import {AppError} from '../infrastructure';
+import {getFactory, get} from '../services/label';
+
+export class RowSelector {
+	constructor(model) {
+		this.model = model;
+	}
+
+	map(items) {
+		const selectionState = this.model.selection();
+		switch (selectionState.unit) {
+			case 'row':
+				return this.mapFromRows(items);
+			case 'column':
+				return this.mapFromColumns(items);
+			case 'cell':
+				return this.mapFromCells(items);
+			case 'mix':
+				return this.mapFromMix(items);
+			default:
+				throw new AppError('row.selector', `Invalid unit ${selectionState.unit}`);
+		}
+	}
+
+	mapFromRows(rows) {
+		const result = [];
+		const columns = this.model.view().columns;
+
+		for(const row of rows) {
+			const line = [];
+
+			for (const column of columns) {
+				const label = getFactory(column);
+				const value = label(row);
+				if(value && typeof value === 'string') {
+					line.push(value);
+				}
+			}
+
+			result.push(line);
+		}
+
+		return result;
+	}
+
+	mapFromColumns(columns) {
+		const result = [];
+		const rows = this.model.view().rows;
+
+		columns.forEach((column, columnIndex) => {
+			const label = getFactory(column);
+			const cells = rows.map(row => label(row));
+
+			if (!result.length) {
+				cells.forEach((cell) => result.push(['empty']));
+			}
+
+			cells.forEach((cell, cellIndex) => {
+				result[cellIndex][columnIndex] = cells[cellIndex];
+			})
+
+		});
+
+		return result;
+	}
+
+	mapFromCells(items) {
+		const result = [];
+		let line = [];
+
+		let keysForComparison = [];
```

Comment:
what is the purpose for that?

Reply:
it helps accumulate keys of columns for further rows(lines) separation

Reply:
it's more convenient to group by cells by rows order by row indices and fill by columns